### ISSUE TITLE
Can't build test code from official tutorial (error: ‘spl_ce_Countable’ undeclared)

### ISSUE THREAD

**Issue by cayolblake** [zephir-lang/zephir#2361]
On PHP 8.1,

Following https://docs.zephir-lang.com/0.12/en/tutorial

```
$ php81 ~/bin/zephir init utils

$ php81 ~/bin/zephir build

 Preparing for PHP compilation...
 Preparing configuration file...
 Compiling...
 Zephir version has changed, use "zephir fullclean" to perform a full clean of the project
 Installing...

 [ERROR] Internal extension compilation failed. Check compile-errors.log for more information.

```

And the file `compile-errors.log`

```
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count’:
~/zephir/utils/ext/kernel/main.c:154:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  154 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c:154:59: note: each undeclared identifier is reported only once for each function it appears in
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count_ev’:
~/zephir/utils/ext/kernel/main.c:204:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  204 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count_int’:
~/zephir/utils/ext/kernel/main.c:252:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  252 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_function_exists’:
~/zephir/utils/ext/kernel/main.c:285:108: warning: comparison between pointer and integer
  285 |         if (zend_hash_str_exists(CG(function_table), Z_STRVAL_P(function_name), Z_STRLEN_P(function_name)) != NULL) {
      |                                                                                                            ^~
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_function_exists_ex’:
~/zephir/utils/ext/kernel/main.c:301:83: warning: comparison between pointer and integer
  301 |         if (zend_hash_str_exists(CG(function_table), function_name, function_len) != NULL) {
      |                                                                                   ^~
make: *** [Makefile:208: kernel/main.lo] Error 1
make: *** Waiting for unfinished jobs....
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count’:
~/zephir/utils/ext/kernel/main.c:154:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  154 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c:154:59: note: each undeclared identifier is reported only once for each function it appears in
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count_ev’:
~/zephir/utils/ext/kernel/main.c:204:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  204 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_fast_count_int’:
~/zephir/utils/ext/kernel/main.c:252:59: error: ‘spl_ce_Countable’ undeclared (first use in this function); did you mean ‘zend_ce_countable’?
  252 |                 if (instanceof_function(Z_OBJCE_P(value), spl_ce_Countable)) {
      |                                                           ^~~~~~~~~~~~~~~~
      |                                                           zend_ce_countable
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_function_exists’:
~/zephir/utils/ext/kernel/main.c:285:108: warning: comparison between pointer and integer
  285 |         if (zend_hash_str_exists(CG(function_table), Z_STRVAL_P(function_name), Z_STRLEN_P(function_name)) != NULL) {
      |                                                                                                            ^~
~/zephir/utils/ext/kernel/main.c: In function ‘zephir_function_exists_ex’:
~/zephir/utils/ext/kernel/main.c:301:83: warning: comparison between pointer and integer
  301 |         if (zend_hash_str_exists(CG(function_table), function_name, function_len) != NULL) {
      |                                                                                   ^~
make: *** [Makefile:208: kernel/main.lo] Error 1
```

And, upon checking, it appears like it might be a function name issue in the generated `ext/kernel/main.c` file.

What do you think?

**Comment by cayolblake** [comment]
So, it appears the latest `zephir.phar` file from here https://github.com/zephir-lang/zephir/releases, which is also from the tutorials, does not support PHP 8.0 or 8.1.

What I did, unfortunately, is...

```
git clone https://github.com/zephir-lang/zephir
cd zephir
composer install
```

Then I linked to the `zephir` wrapper script from within the dir and used that for init, generate, and compile moving on.

Couldn't find, or at least in a straightforward way, how to create a combined `zephir.phar` file like the ones distributed.
